eZ/Publish/Core/Search/Legacy/Content/Gateway/DoctrineDatabase.php

```diff
@@ -176,8 +180,12 @@ protected function getResultCount(Criterion $filter, $fieldFilters)
     protected function getContentInfoList(Criterion $filter, $sort, $offset, $limit, $fieldFilters)
     {
         $query = $this->handler->createSelectQuery();
-        $query->selectDistinct(
+        $query->select(
             'ezcontentobject.*',
+            // Content object names
+            $this->handler->aliasedColumn($query, 'name', 'ezcontentobject_name'),
+            $this->handler->aliasedColumn($query, 'content_translation', 'ezcontentobject_name'),
+            // Main node id (if there is one)
```

Comment:
The whole change here is breaking paging. as content objects will take several rows.